scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2888,6 +2894,36 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     }
   }
 
+  def caseDef(mods: List[Mod]): Stat = {
+    accept[KwCase]
+
+    // Scanner thinks it is in a pattern match after seeing the `case`.
+    // We need to get it out of that mode by telling it we are past the `=>`
+    adjustSepRegions(Token.RightArrow(input, dialect, 0, 0))
+
+    if (ahead(token.is[Comma])) {
+      val cases = commaSeparated(readCase())
+      Defn.Enum.RepeatedCase(mods, cases)
+    } else {
+      val caseName = termName()
+      val typeParams = typeParamClauseOpt(ownerIsType = true, ctxBoundsAllowed = true)
+      val ctor = primaryCtor(OwnedByClass)
+      val parents = if (token.is[KwExtends]) {
+        accept[KwExtends]
+        templateParents()
+      } else {
+        Nil
+      }
+      Defn.Enum.Case(mods, caseName, typeParams, ctor, parents)
+    }
+  }
+
+  def readCase(): Defn.Enum.Name = {
```

Comment:
I think you should use the `def name` parser. I suggest adding a new `enumName` parser:
```scala
def enumName(): Term.Name = name(Enum.Name(_), advance = true)
```
Similar to:
https://github.com/scalameta/trees/blob/527cfabf82fad0e55d6af189517b822656735039/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala#L1158-L1159